### DISCUSSION TITLE
"Segmentation fault (core dumped) $runcommand > /dev/null 2>&1" occured when running DOSXYZnrc with egs-parallel

### DISCUSSION THREAD

**Discussion by JamsNeroc** [nrc-cnrc/EGSnrc#1079] [Questions] [answered]
"Segmentation fault (core dumped) $runcommand > /dev/null 2>&1" occured when running DOSXYZnrc with egs-parallel
Hello everyone,
I'm using BEAMnrc/DOSXYZnrc egs-parallel to simulate a treatment head and a CT phantom. A error of core dump would occur when using 1e7 and more histories in DOSXYZnrc (see below), but the simulation could exit normally and genertare .3ddose file when using 1e6 histories. Is there any ways can solve this problem?

`geant4@geant4:~/EGSnrc/egs_home/dosxyznrc$ egs-parallel --batch cpu -q long -n16 -f -v -c 'dosxyznrc -i zMCphantomARC_dosxyznrc.egsinp -p 521icru'
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.384557498: BEGIN egs-parallel
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.389381661: EGSnrc environment:
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.394245653:     HEN_HOUSE  = /home/geant4/EGSnrc/HEN_HOUSE/
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.398981590:     EGS_HOME   = /home/geant4/EGSnrc/egs_home/
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.403754077:     EGS_CONFIG = /home/geant4/EGSnrc/HEN_HOUSE/specs/egs.conf
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.434187133: parallel options:
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.440739923:     batch      = cpu
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.447042633:     queue      = long
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.452598860:     nthread    = 16
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.458704176:     delay      = 0
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.464196839:     command    = dosxyznrc -i zMCphantomARC_dosxyznrc.egsinp -p 521icru
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.470482641:     basename   = zMCphantomARC_dosxyznrc
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.475505664:     first job  = 1
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.482122437:     options    = 
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.490818368: log file: /home/geant4/EGSnrc/egs_home/dosxyznrc/zMCphantomARC_dosxyznrc.egsparallel
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.496066070: cd /home/geant4/EGSnrc/egs_home/dosxyznrc
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.500902738: EXEC egs-parallel-cpu long 16 0 1 zMCphantomARC_dosxyznrc 'dosxyznrc -i zMCphantomARC_dosxyznrc.egsinp -p 521icru' '' verbose
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.536457413: BEGIN /home/geant4/EGSnrc/HEN_HOUSE/scripts/egs-parallel-cpu
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.561729089: BEGIN host=geant4
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.566485192: job 0001: RUN dosxyznrc -i zMCphantomARC_dosxyznrc.egsinp -p 521icru -b -P 16 -j 1 -f 1
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.573666038: job 0001: host=geant4 pid=25459
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.581376793: job 0002: RUN dosxyznrc -i zMCphantomARC_dosxyznrc.egsinp -p 521icru -b -P 16 -j 2 -f 1
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.588091163: job 0002: host=geant4 pid=25477
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.596668399: job 0003: RUN dosxyznrc -i zMCphantomARC_dosxyznrc.egsinp -p 521icru -b -P 16 -j 3 -f 1
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.605210796: job 0003: host=geant4 pid=25488
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.613480188: job 0004: RUN dosxyznrc -i zMCphantomARC_dosxyznrc.egsinp -p 521icru -b -P 16 -j 4 -f 1
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.620550034: job 0004: host=geant4 pid=25501
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.628703226: job 0005: RUN dosxyznrc -i zMCphantomARC_dosxyznrc.egsinp -p 521icru -b -P 16 -j 5 -f 1
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.636124411: job 0005: host=geant4 pid=25511
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.646096213: job 0006: RUN dosxyznrc -i zMCphantomARC_dosxyznrc.egsinp -p 521icru -b -P 16 -j 6 -f 1
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.655306249: job 0006: host=geant4 pid=25523
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.669746727: job 0007: RUN dosxyznrc -i zMCphantomARC_dosxyznrc.egsinp -p 521icru -b -P 16 -j 7 -f 1
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.678266410: job 0007: host=geant4 pid=25536
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.687557958: job 0008: RUN dosxyznrc -i zMCphantomARC_dosxyznrc.egsinp -p 521icru -b -P 16 -j 8 -f 1
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.698071938: job 0008: host=geant4 pid=25548
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.706496384: job 0009: RUN dosxyznrc -i zMCphantomARC_dosxyznrc.egsinp -p 521icru -b -P 16 -j 9 -f 1
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.722193009: job 0009: host=geant4 pid=25560
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.733580760: job 0010: RUN dosxyznrc -i zMCphantomARC_dosxyznrc.egsinp -p 521icru -b -P 16 -j 10 -f 1
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.742991886: job 0010: host=geant4 pid=25574
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.752479292: job 0011: RUN dosxyznrc -i zMCphantomARC_dosxyznrc.egsinp -p 521icru -b -P 16 -j 11 -f 1
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.761352046: job 0011: host=geant4 pid=25588
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.772611447: job 0012: RUN dosxyznrc -i zMCphantomARC_dosxyznrc.egsinp -p 521icru -b -P 16 -j 12 -f 1
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.781489721: job 0012: host=geant4 pid=25600
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.804541923: job 0013: RUN dosxyznrc -i zMCphantomARC_dosxyznrc.egsinp -p 521icru -b -P 16 -j 13 -f 1
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.815487989: job 0013: host=geant4 pid=25612
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.865714887: job 0014: RUN dosxyznrc -i zMCphantomARC_dosxyznrc.egsinp -p 521icru -b -P 16 -j 14 -f 1
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.874014594: job 0014: host=geant4 pid=25626
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.911557622: job 0015: RUN dosxyznrc -i zMCphantomARC_dosxyznrc.egsinp -p 521icru -b -P 16 -j 15 -f 1
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:39.918103802: job 0015: host=geant4 pid=25640
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:40.002802228: job 0016: RUN dosxyznrc -i zMCphantomARC_dosxyznrc.egsinp -p 521icru -b -P 16 -j 16 -f 1
EGSnrc egs-parallel 2023-11-24 (UTC) 07:39:40.068727365: job 0016: host=geant4 pid=25653
/home/geant4/EGSnrc/HEN_HOUSE/scripts/egs-parallel-cpu: line 156: 25511 Segmentation fault      (core dumped) $runcommand > /dev/null 2>&1
/home/geant4/EGSnrc/HEN_HOUSE/scripts/egs-parallel-cpu: line 156: 25612 Segmentation fault      (core dumped) $runcommand > /dev/null 2>&1
EGSnrc egs-parallel 2023-11-24 (UTC) 07:53:25.774950930: DONE.
`

**Comment by rtownson** [comment]
Could you post the .egslog file from one of the jobs? Are you using a phase-space source from BEAMnrc, or a shared library source? Is the voxel grid very high resolution, is there any chance that the computer is running out of memory?

**Reply by JamsNeroc** [reply]
Hello @rtownson, thanks for your reply!
I have uploaded three .egslog files where *_w1.egslog is the job exited normally, *_w9.egslog and *_w10.egslog are the jobs not exited normally.  The file extension had been changed to .txt for uploading.
[zMCphantomARC_DOSXYZnrc_w1.txt](https://github.com/nrc-cnrc/EGSnrc/files/13463484/zMCphantomARC_DOSXYZnrc_w1.txt)
[zMCphantomARC_DOSXYZnrc_w9.txt](https://github.com/nrc-cnrc/EGSnrc/files/13463485/zMCphantomARC_DOSXYZnrc_w9.txt)
[zMCphantomARC_DOSXYZnrc_w10.txt](https://github.com/nrc-cnrc/EGSnrc/files/13463486/zMCphantomARC_DOSXYZnrc_w10.txt)

I'm using "Source 21 - Dynamic BEAM simulation source with multiple variable geometry setting" in DOSXYZnrc, and a shared library source, which was created in BEAMnrc, was used in the simulation. A screenshot has been posted below.
![1700905063704](https://github.com/nrc-cnrc/EGSnrc/assets/63440063/bf367790-c3e3-4b2f-8dcd-5cbd1360419e)

The voxel grid is (2.5, 2.5, 5)mm, and I don't think that the computer is running out of memory because I had done a same simluation in a workstation which has a 128 RAM and the same core dump occured too.

I can upload my .egsinp files of BEAMnrc\DOSXYZnrc and the .egsphant file if you think it's necessary.
Looking forward to your reply.

**Reply by rtownson** [reply, marked as answer]
I have seen this issue before when using vacuum as the dsurround material. Try changing the surrounding material to air instead of vacuum. You'll having to do this manually in the .egsinp file I believe, changing a '0' to a '1' (since air is the first material in your list for the egsphant file). The parameter to edit is `medsur`, right before `dsurround(1)`. See page 29 in the dosxyznrc user manual. You can spot the number easily by looking for the `50`:
```
2           0           0  50.000           0
```
should be
```
2           0           1  50.000           0
```
Also, make sure that 50cm is actually enough padding to including the source from all possible angles. You may need to increase it.

**Reply by JamsNeroc** [reply]
I'm happy to tell you that the egs-parallel could exit normally after I changed the surrounding material to air instead of vacuum in DOSXYZnrc. It's hard for me to notice this tiny detail. 
Thanks for your answer, it's really helpful.
